cli/azd/pkg/project/framework_service_none.go

```diff
@@ -0,0 +1,75 @@
+package project
+
+import (
+	"context"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/async"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+)
+
+type NoneProject struct {
+}
+
+func NewNoneProject() FrameworkService {
+	return &NoneProject{}
+}
```

Comment:
This is confusing.  Are we adding a NoneProject to satisfy azd-current-implementation?  Ideally, implementation details should not drive the design, but the other way around.

Is there any other alternative where we don't need to have such `FrameworkService` (None) ?

I'm guessing this is a hack to enable a service with `host: containerapps` to omit the language and bypass/null the packaging step. 
Can we ignore the frameworkService from the execution-flow when `aca` is the host ? 
It feels easier to see a design/condition for adding/injecting a frameworkService based on the language+host than looking at an frameworkService called None and understanding why it is there.

What happen, for example, if the host is not aca and the language is not defined?  That should still be an error, right?

Reply:
Similar to my comment above - once we start bringing/referencing prebuilt/public containers/charts we lose the requirement of requiring project code that has a language.

Today service targets that `azd` supports that target containers would be ACA and AKS. (and even app service but we don't support this config today)

I did think about not having a `NoneProject` but this then leads us down the path of always checking for the existence of a framework service before calling it.  The `NoneProject` essentially is a `noop` but allows us to more easily work in our existing framework.

I'll think on this a bit more but this is where I initially landed.

Reply:
Have this `NoneProject` implementation also allows us to keep all the rules for what `none` means in a central location/implementation instead of having to scatter the business rules across multiple components..

Reply:
My preference when it comes to SW is to avoid creating components or abstractions which only makes sense after knowing about how the system works internally but are non-sense by itself.

Here's an example:
a banking system with automatic weekly payment to workers depending on the hours they work.  Imagine there is a component that represents a payment deposit of $0 for the case then the worker didn't work any hour. The system internal implementation requires to create a check and mail it to the worker, so the component ensures the system will just send a $0 check to the worker.  

In such example, the system should not send a check and have a different behavior by design.

So, I see it similar, we are introducing a component to represent something with no meaning by itself just to make internal things to keep working, instead of updating the system to handle the case by design.

So, I would leave this to you and @ellismg to decide.  My suggestion is to make the `host` attribute from a service a `first class attribute` which the system uses to know what operations are allowed and should run or not.

Reply:
I just noticed that @ellismg already had a `NoopProject` that satisfied the same requirements. Going to deprecate mine in favor of this.